eval/src/vespa/eval/instruction/sparse_dot_product_function.cpp

```diff
@@ -13,41 +13,45 @@ using namespace instruction;
 
 namespace {
 
-template <typename SCT, typename BCT>
-double my_fast_sparse_dot_product(const FastValueIndex &small_idx, const FastValueIndex &big_idx,
-                                  const SCT *small_cells, const BCT *big_cells)
+template <typename CT, bool single_dim>
+double my_fast_sparse_dot_product(const FastValueIndex *small_idx, const FastValueIndex *big_idx,
```

Comment:
it looks like this method only uses foo_idx->map, so it could take small_map and big_map instead?